src/reflect/scala/reflect/internal/Symbols.scala

```diff
@@ -2385,7 +2385,13 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
       } else Nil
     }
 
-    /** Equivalent to allOverriddenSymbols.nonEmpty, but more efficient. */
+    /** Equivalent to allOverriddenSymbols.nonEmpty, but more efficient.
+      *
+      * Drive-by warning: the result of `allOverriddenSymbols` can go from
+      * non-empty to empty following `erasure`; it seems that this is always
+      * initialized pre-erasure, so post-erasure this method is no longer
+      * equivalent to `allOverriddenSymbols.nonEmpty` like it claims.
+      */
     lazy val isOverridingSymbol = (
          canMatchInheritedSymbols
       && owner.ancestors.exists(base => overriddenSymbol(base) != NoSymbol)
```

Comment:
Another drive-by warning: this cache can serve up outdated values in multi-Run compilation if the ancestors change! Other caches in `Symbol`. We should store either the `Run` or maybe the full `Period` alongside this boolean and recompute if needed.

Reply:
I'll put that in a separate commit once I deal with the `ComputeBridges` suggestions.

Reply:
I wound up just caching `allOverriddenSymbols`...

It appears that `isOverridingSymbol` gets called (after `typer`) mainly in [`constructors`](https://github.com/scala/scala/blob/e100de3e8c82b7e1812c2ea47cd31fc17be5539a/src/compiler/scala/tools/nsc/transform/Constructors.scala#L168) and [`explicitouter`](https://github.com/scala/scala/blob/e100de3e8c82b7e1812c2ea47cd31fc17be5539a/src/compiler/scala/tools/nsc/transform/ExplicitOuter.scala#L456), and probably at most once per symbol per each of those phases, so I don't feel _that_ bad about moving the cache to `allOverriddenSymbols`.